### ISSUE TITLE
Allow per-channel configuration options

### ISSUE THREAD

**Issue by njern** [nats-io/nats-streaming-server#164]
Several of the options passed  to the server (max_subs, max_msgs, max_bytes) are options I'd like to be able to define per channel and I'm sure this would be useful to other users as well. 

Is this something that has been considered already? Any thoughts on where it would fit in terms of configuration? E.g on server start, defined by the publisher when a channel is first created or somewhere else I have not thought of?

**Comment by kozlovic** [comment]
Yes, this is something that has been discussed. We were thinking of a configuration file where you would define the channels with their limits. What do you think?

**Comment by njern** [comment]
I think that'd work nicely.
